### ISSUE TITLE
Actor-based objects cant serialize `API_FIELD() Array<T>`

### ISSUE THREAD

**Issue by Withaust** [FlaxEngine/FlaxEngine#1040]
**Issue description:**
Currently, if you were going to use a code like this:
```cpp
API_FIELD() Array<Actor*> Spawns;
```
in an actor-based object, then it will never serialize changes to the container in the file. When it comes to script-based objects, then everything seems to be working correctly. 

Here on the left side you can see this code being used by script-based object attached to the actor-based object and filled with the same values, while on the right side you can see that only the script actually serialized and wrote something to the scene file:
![image](https://user-images.githubusercontent.com/118038102/233862252-53cbbeaa-9352-4bd9-91d3-59e54e76eb22.png)

**Steps to reproduce:**
Try using `API_FIELD() Array<T>` in an Actor-based object

**Flax version:**
Flax 1.5.6338 on Windows 10 x64

**Comment by mafiesto4** [comment]
Did you add `API_AUTO_SERIALIZATION();` tag in that custom actor class? If not, then fields/properties need manual serialization on C++ side via `ISerializable` interface methods override.

Docs about C++ serialziation: https://docs.flaxengine.com/manual/scripting/cpp/serialization.html

**Comment by Tryibion** [comment]
I just tried this using `API_FIELD() Array<Actor*> actors` with `API_AUTO_SERIALIZATION();` 

```
[ 00:00:44.643 ]: [Info] Z:\Flax\Source\FlaxEngine\Source\Engine\Serialization\Serialization.h(459): error C3861: 'FindObject': identifier not found
Z:\Flax\Source\FlaxEngine\Source\Engine\Serialization\Serialization.h(597): note: see reference to function template instantiation 'void Serialization::Deserialize<Actor>(ISerializable::DeserializeStream &,T *&,ISerializeModifier *)' being compiled
        with
        [
            T=Actor
        ]
C:\Users\mrcha\Documents\Flax Projects\TestingProject\Cache\Intermediate\GameEditorTarget\Windows\x64\Development\Game\Game.Bindings.Gen.cpp(30): note: see reference to function template instantiation 'void Serialization::Deserialize<Actor*,HeapAllocation>(ISerializable::DeserializeStream &,Array<Actor *,HeapAllocation> &,ISerializeModifier *)' being compiled
[ 00:00:44.649 ]: [Info] Task C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.35.32215\bin\HostX64\x64\cl.exe /nologo /c /std:c++14 /Oi /Gy /analyze- /Zc:inline /Ot /Ob2 /Zi /Zo /Ox /Oi /Oy /FC /errorReport:prompt /EHsc /GF /MD /fp:fast /fp:except- /GS /GR /WX- /W3 /wd"4005" /Zc:wchar_t /D "PLATFORM_WIN32" /D "WIN32" /D "_CRT_SECURE_NO_DEPRECATE" /D "_CRT_SECURE_NO_WARNINGS" /D "_WINDOWS" /D "WIN64" /D "PLATFORM_WINDOWS" /D "UNICODE" /D "_UNICODE" /D "USE_EDITOR" /D "BUILD_DEVELOPMENT" /D "NDEBUG" /D "FLAX_1" /D "FLAX_1_5" /D "FLAX_1_1_OR_NEWER" /D "FLAX_1_2_OR_NEWER" /D "FLAX_1_3_OR_NEWER" /D "FLAX_1_4_OR_NEWER" /D "FLAX_1_5_OR_NEWER" /D "GAME_API=__declspec(dllexport)" /D "TESTPLUGIN_API=__declspec(dllimport)" /D "SIMPLESAVESYSTEM_API=__declspec(dllimport)" /D "FLAXENGINE_API=__declspec(dllimport)" /D "COMPILE_WITH_PROFILER" /D "TRACY_ENABLE" /D "TRACY_ON_DEMAND" /D "COMPILE_WITH_DEBUG_DRAW" /D "COMPILE_WITH_SHADER_CACHE_MANAGER" /D "COMPILE_WITH_MODEL_TOOL" /D "COMPILE_WITH_SHADER_COMPILER" /D "COMPILE_WITH_CSG_BUILDER" /D "COMPILE_WITH_NAV_MESH_BUILDER" /D "COMPILE_WITH_NETWORKING" /D "COMPILE_WITH_PHYSICS_COOKING" /D "COMPILE_WITH_PARTICLE_GPU_GRAPH" /D "COMPILE_WITH_GPU_PARTICLES" /I"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.35.32215\include" /I"C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt" /I"C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\shared" /I"C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\um" /I"C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\winrt" /IZ:\Flax\Source\FlaxEngine\Source\ThirdParty\mono-2.0 /IZ:\Flax\Source\FlaxEngine\Source\ThirdParty /I"C:\Users\mrcha\Documents\Flax Projects\TestingProject\Source" /IZ:\Flax\Source\FlaxEngine\Source /I"C:\Users\mrcha\Documents\Flax Projects\TestingProject\Plugins\TestPlugin\Source" /I"C:\Users\mrcha\Documents\Flax Projects\TestingProject\Plugins\SimpleSaveSystem\Source" /Fd"C:\Users\mrcha\Documents\Flax Projects\TestingProject\Cache/Intermediate\GameEditorTarget\Windows\x64\Development\Game\Game.Bindings.Gen.pdb" /Fo"C:\Users\mrcha\Documents\Flax Projects\TestingProject\Cache/Intermediate\GameEditorTarget\Windows\x64\Development\Game\Game.Bindings.Gen.obj" "C:\Users\mrcha\Documents\Flax Projects\TestingProject\Cache/Intermediate\GameEditorTarget\Windows\x64\Development\Game\Game.Bindings.Gen.cpp" failed with exit code 2

[ 00:00:44.681 ]: [Info] 2 tasks failed
```

**Comment by Withaust** [comment]
@Tryibion I used this variant of the code
```

API_CLASS(Attributes = "ActorContextMenu(\"New/Other/Test\"), ActorToolbox(\"Other\")")
class GAME_API Test : public Actor
{
    API_AUTO_SERIALIZATION();
    DECLARE_SCRIPTING_TYPE(Test);

public:

    API_FIELD() Array<Actor*> Spawns;

    void OnEnable() override;
    void OnDisable() override;
};
```
and it seems to work fine for me, everything is serialized properly now.

**Comment by Tryibion** [comment]
I tried a script and an actor and I get the same scripts compilation error.

**Comment by mafiesto4** [comment]
This problem with compilation should be fixed in 5d2a3482c770abb9ee522d6bd39dce1a62f77cfc
